lib/GADS.pm

```diff
@@ -2753,6 +2753,235 @@ prefix '/:layout_name' => sub {
         template 'data' => $params;
     };
 
+    prefix '/report' => sub {
+
+        #view all reports for this instance
+        get '' => require_login sub {
+            my $user   = logged_in_user;
+            my $layout = var('layout') or pass;
+
+            my %params = (
+                user   => $user,
+                layout => $layout,
+                schema => schema,
+            );
+
+            my $records = GADS::Records->new(%params);
+
+            my $alert = GADS::Alert->new(
+                user   => $user,
+                layout => $layout,
+                schema => schema,
+            );
+
+            my $base_url = request->base;
+
+            my $params;
+
+            $params->{alerts}      = $alert->all;
+            $params->{header_type} = 'table_tabs';
+
+            $params->{layout_obj} = $layout;
+            $params->{layout}     = $layout;
+
+            $params->{header_back_url} = "${base_url}table";
+            $params->{breadcrumbs}     = [
+                Crumb( $base_url . "table/", "Tables" ),
+                Crumb( "",                   "Table: " . $layout->name )
+            ];
+
+            my $layout_id = $layout->{instance_id};
+
+            my $reports = $layout->reports;
+
+            $params->{reports} = $reports;
+
+            template 'reports/view' => $params;
+        };
+
+        #add a report
+        any [ 'get', 'post' ] => '/add' => require_login sub {
+            if ( body_parameters && body_parameters->{submit} ) {
+                my $layout             = var('layout') or pass;
+                my $report_description = params->{'report_description'};
+                my $report_name        = params->{'report_name'};
+                my $checkbox_fields    = params->{'checkboxes'};
+                my $instance           = $layout->{instance_id}
+                  if $layout && $layout->{instance_id};
```

Comment:
Please access instance_id as a method (i.e. `$layout->instance_id`) as using a hash will not always work correctly. Also, I don't think there needs to be an `if` here as `$layout` will always be true.

Reply:
The if was there more as a "just making sure I didn't break it" statement - I like to make sure I've not missed anything. I'll remove :-)